tests/integration/framework/framework.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework/iowriter"
+	"github.com/dapr/dapr/tests/integration/framework/kill"
+)
+
+// daprdOptions contains the options for running Daprd in integration tests.
+type daprdOptions struct {
+	stdout io.WriteCloser
+	stderr io.WriteCloser
+
+	binPath string
+	appID   string
+
+	appPort                 int
+	grpcPort                int
+	httpPort                int
+	internalGRPCPort        int
+	publicPort              int
+	metricsPort             int
+	profilePort             int
+	runErrorFn              func(error)
+	exitCode                int
+	appHealthCheck          bool
+	appHealthCheckPath      string
+	appHealthProbeInterval  int
+	appHealthProbeThreshold int
+}
+
+// RunDaprdOption is a function that configures the DaprdOptions.
+type RunDaprdOption func(*daprdOptions)
+
+type Command struct {
+	lock sync.Mutex
+	cmd  *exec.Cmd
+
+	cmdcancel    context.CancelFunc
+	runErrorFnFn func(error)
+	exitCode     int
+	stdoutpipe   io.WriteCloser
+	stderrpipe   io.WriteCloser
+
+	AppID            string
+	AppPort          int
+	GRPCPort         int
+	HTTPPort         int
+	InternalGRPCPort int
+	PublicPort       int
+	MetricsPort      int
+	ProfilePort      int
+}
+
+func RunDaprd(t *testing.T, ctx context.Context, opts ...RunDaprdOption) *Command {
+	t.Helper()
+
+	uid, err := uuid.NewUUID()
+	require.NoError(t, err)
+
+	appListener, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
Can we use a UNIX socket as listener? It seems that every time we use an actual TCP socket in tests, we get flakiness.

Reply:
Actually.... *no* 😱

You can configure Dapr to listen on a UDS (`--unix-domain-socket`) for its APIs. But you cannot make Dapr use a UDS to talk to an app.

This seems to be a missing piece.